src/App.js

```diff
@@ -14,11 +16,118 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  showList = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  }
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortBy: 'alphabet',
+    });
+  }
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  }
+
+  render() {
+    const { goods, isVisible, isReversed, sortBy } = this.state;
+    const sortedGoods = goods.filter(item => item);
```

Comment:
```suggestion
    const sortedGoods = [...goods]
```
There's a better way to make a copy of an array